sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubMessageToRow.java

```diff
@@ -226,11 +260,11 @@ public void processElement(
                             field, timestamp, element.getAttributeMap(), payload))
                 .collect(toList());
         o.get(MAIN_TAG).output(Row.withSchema(messageSchema).addValues(values).build());
-      } catch (UnsupportedRowJsonException jsonException) {
+      } catch (UnsupportedRowJsonException | AvroRuntimeException exception) {
```

Comment:
I think it would be preferable to define our own `ParseException` that parsePayload throws. Then parsePayload can catch these exceptions specific to the format that is being used, and throw it with the format-specific exception as the cause. That way the calls to parsePayload just need to check for `ParseException`

Reply:
Done